src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1779,18 +1822,89 @@ RelationRestrictionPartitionKeyIndex(RelationRestriction *relationRestriction)
 
 		if (!targetEntry->resjunk &&
 			IsA(targetExpression, Var) &&
-			IsPartitionColumn(targetExpression, relationPlannerParseQuery))
+			IsPartitionColumn(targetExpression, originalQueryContainingRTEIdentity))
 		{
 			Var *targetColumn = (Var *) targetExpression;
 
-			if (targetColumn->varno == relationRestriction->index)
+/*
+ * As of pg13, columns point to RELATION_RTE so we don't need to do
```

Comment:
why did we remove pg 13 ifdefs here?

Reply:
ping on this

Reply:
Because now we need to process the varlevelsup > 0 even for pg13. Otherwise we will have the same regression on master compared to this branch on pg13.

Adding the ifdefs back makes 2 tests not pushdown on pg13.

Reply:
So, the comment is wrong and also wrongly indented :) 

I think we should tell here that we have to do this to support distributions columns that are correlated